reframe/core/schedulers/slurm.py

```diff
@@ -157,6 +157,8 @@ def _count_compatible_nodes(self, nodes):
         constraints = set()
         partitions = (set(self.sched_partition.split())
                       if self.sched_partition else set())
+        excluded_node_names = ({n.name for n in self._get_excluded_nodes()}
+                               if self.sched_exclude_nodelist else set())
```

Comment:
Why don't you incorporate this logic in the `_get_excluded_nodes()` so as to make this statement much simpler?